packages/mdc-menu/foundation.ts

```diff
@@ -97,8 +97,10 @@ export class MDCMenuFoundation extends MDCFoundation<MDCMenuAdapter> {
 
     // Wait for the menu to close before adding/removing classes that affect styles.
     this.closeAnimationEndTimerId_ = setTimeout(() => {
-      if (this.adapter_.isSelectableItemAtIndex(index)) {
-        this.setSelectedIndex(index);
+      // Recompute the index in case the menu contents have changed.
```

Comment:
I think, the list foundation need to have a separate method (For example, `.layout()`) to recompute indexes which can be called on demand.